### ISSUE TITLE
Allow registering Service Worker URL without `https` when using localhost

### ISSUE THREAD

**Issue by miina** [xwp/pwa-wp#49]
Currently `https` is always enforced when registering the Service Worker ([see here](https://github.com/xwp/pwa-wp/blob/master/wp-includes/service-workers.php#L56)), however, `http` is allowed on `localhost` according to [this page](https://developers.google.com/web/fundamentals/primers/service-workers/#you_need_https):
> During development you'll be able to use service worker through localhost, but to deploy it on a site you'll need to have HTTPS setup on your server.

When developing on `localhost` without using `https` the Service Worker is not registered and an error shown in console instead, for example:
![screen shot 2018-08-06 at 9 17 15 pm](https://user-images.githubusercontent.com/3294597/43734251-cd4ea488-99bf-11e8-805e-6fce51312ddf.png)

Perhaps we should consider registering the Service Worker without enforcing the `https` in case of `localhost` to make it easier for developers using `http` locally on `localhost`.

**Comment by westonruter** [comment]
Yes, that is good. In the case of `localhost` or `localhost:port` then the scheme could be omitted to re-use the default.
